server/sandbox_stop_linux.go

```diff
@@ -63,7 +63,7 @@ func (s *Server) stopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 			if err := s.Runtime().StopContainer(ctx, c, timeout); err != nil {
 				return nil, fmt.Errorf("failed to stop container %s in pod sandbox %s: %v", c.Name(), sb.ID(), err)
 			}
-			if err := s.Runtime().WaitContainerStateStopped(ctx, c, timeout); err != nil {
+			if err := s.Runtime().WaitContainerStateStopped(ctx, c); err != nil {
```

Comment:
Only question I have is if the kubelet going to think that the entire operation failed if we exceed the timeout passed by the kubelet.

Reply:
Kubelet is passing the timeout, right? It's up to the CRI implementation to actually apply this timeout value, so I don't think kubelet will think the operation has failed. I'm not sure about this, let me know if I am wrong here!